src/main/java/com/minecolonies/coremod/client/gui/WindowHutBuilder.java

```diff
@@ -76,21 +135,57 @@ public int getElementCount()
             {
                 return resources.size();
             }
-
             @Override
             public void updateElement(final int index, @NotNull final Pane rowPane)
             {
-                final Object obj = entries[index];
+                    final BuildingBuilder.BuildingBuilderResource resource = resources.get(index);
+                    Label resourceLabel = rowPane.findPaneOfTypeByID(RESOURCE_NAME, Label.class);
+                    Label neededLabel = rowPane.findPaneOfTypeByID(RESOURCE_AVAILABLE_NEEDED, Label.class);
+                    Button addButton = rowPane.findPaneOfTypeByID(RESOURCE_ADD, Button.class);
+
+                    switch (resource.getAvailabilityStatus())
+                    {
+                        case DONT_HAVE:
+                            addButton.disable();
+                            resourceLabel.setColor(red, red);
+                            neededLabel.setColor(red, red);
+                            break;
+                        case NEED_MORE:
+                            addButton.enable();
+                            resourceLabel.setColor(red, red);
+                            neededLabel.setColor(red, red);
+                            break;
+                        case HAVE_ENOUGHT:
+                            addButton.enable();
+                            resourceLabel.setColor(darkgreen, darkgreen);
+                            neededLabel.setColor(darkgreen, darkgreen);
+                            break;
+                        case NOT_NEEDED:
+                        default:
+                            addButton.disable();
+                            resourceLabel.setColor(black,black);
+                            neededLabel.setColor(black,black);
+                            break;
+
+                    }
+
+                    //position the addRessource Button to the right
+                    final int buttonX = rowPane.getWidth() - addButton.getWidth() - (rowPane.getHeight() - addButton.getHeight()) / 2;
+                    final int buttonY = (rowPane.getHeight() - addButton.getHeight())/2;
+                    addButton.setPosition(buttonX,buttonY);
+
+                    resourceLabel.setLabelText(resource.getName());
+                    final int missing = (resource.getAvailable()-resource.getNeeded());
+                    final String missingItems = (missing>0)?"+"+Integer.toString(missing):Integer.toString(missing);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Add parentheses to make the operator precedence explicit. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS864)